lib/HLSL/DxilContainerReflection.cpp

```diff
@@ -120,6 +120,117 @@ enum class PublicAPI { D3D12 = 0, D3D11_47 = 1, D3D11_43 = 2, Invalid };
   ((D3D_SHADER_VARIABLE_CLASS)(D3D_SVC_INTERFACE_POINTER + 1))
 #endif
 
+// HLSL Change start: define some types earlier so they are complete by the time
+// ~unique_ptr<T> is instantiated for them. Fixes C++23 build errors.
```

Comment:
This is a DXC-specific file and doesn't need the "HLSL Change start" marker since the whole file (and HLSL library folder) is HLSL specific.

```suggestion
// Define some types earlier so they are complete by the time ~unique_ptr<T> is
// instantiated for them. Fixes C++23 build errors.
```